common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1601,3 +1601,52 @@ def can_toggle_course_pacing(self):
           bool: False if the course has already started, True otherwise.
         """
         return datetime.now(UTC()) <= self.start
+
+
+class CourseSummary(object):
+    """
+    A lightweight course summary class, which constructs split/mongo course summary without loading
+    the course. It is used at cms for listing courses to global staff user.
+    """
+    course_info_fields = ['display_name', 'display_coursenumber', 'display_organization']
+
+    def __init__(self, course_locator, display_name="Empty", display_coursenumber=None, display_organization=None):
+        """
+        Initialize and construct course summary
+
+        :param course_locator: CourseLocator object of the course.
+        :param display_name: display name of the course. When you create a course from console, display_name
+            isn't set (course block has no key `display_name`). "Empty" name is returned when we load the course.
+            If `display_name` isn't present in the course block, use the `Empty` as default display name.
+            We can set None as a display_name in Course Advance Settings; Do not use "Empty" when display_name is
+            set to None.
+        :param display_coursenumber: Course number that is specified & appears in the courseware
+        :param display_organization: Course organization that is specified & appears in the courseware
```

Comment:
Please use the Google/Napolean format for docstrings. https://github.com/edx/edx-platform/wiki/Python-Guidelines#follow-pep-257